internal/pkg/manifest/svc.go

```diff
@@ -479,7 +479,7 @@ func (rr RoutingRuleConfiguration) exposedPorts(exposedPorts []ExposedPort, work
 // exportPorts returns any new ports that should be exposed given the network load balancer
 // configuration that's not part of the existing containerPorts.
 func (cfg NetworkLoadBalancerListener) exposedPorts(exposedPorts []ExposedPort, workloadName string) ([]ExposedPort, error) {
-	if cfg.IsEmpty() {
+	if cfg.Port == nil {
```

Comment:
The reason I changed it to `Port != nil` as it is consistent with all the other exposedPorts functions. Also, another main important reason why changed is because Main listener is comprised of `NetworkLoadBalancerListener` properties and `Aliases` which makes it important for us to make sure customer has not provided any property to declare Main listener as empty and `cfg.IsEmpty()` doesn't give that answer and `cfg.Port == nil` perfectly does it's job for every NLB listener irrespective of `exopsedPorts()` called before or after validation.

Reply:
> it is consistent with all the other exposedPorts functions.

Do you mean for example L458?
```
if rr.TargetPort == nil {
   return nil
}
```

`rr.TargetPort == nil` makes sense to me, because `exposedPort` returns of a list of container ports exposed by `targetPort`.

However, `cfg.Port == nil` is a little counterintuitive, because `cfg.Port` is the listener port, not the target port. Although `exportPorts` indeed uses  `cfg.Port`, but it's only when `cfg.TargetPort` is `nil`. 

I know that in the real world,  we require `cfg.Port` to be `non-nil`, i.e. there won't be a case when `cfg.TargetPort != nil && cfg.Port == nil`. However, this is not a logic that `exposedPorts` handles.

> Main listener is comprised of `NetworkLoadBalancerListener` properties and `Aliases`

I am confused here. I thought the "main listener" **is** a `NetworkLoadBalancerListener`, and does not contain `Aliases`. `Aliases` is a property of `NetworkLoadBalancerConfiguration`, but this function's receiver is `NetworkLoadBalancerListener`.

Reply:
Ohh you are right, I got confused with the naming. Yeah "main listener" is a NetworkLoadBalancerListener, and does not contain Aliases. So the thing is if customer defines just `nlb.Stickiness: true` then  `if cfg.IsEmpty()` false and we execute rest of the logic for `exposedPorts()` thought ideally we should catch this logic in validate itself. For now we definitely execute validate before we execute `exposedPorts()` so we are safe. But `cfg.Port == nil` keeps us safe in all the conditions. I am fine either way.